.github/workflows/ci.yml

```diff
@@ -6,106 +6,95 @@ on:
         run: echo "SHA7=$(git rev-parse --short ${{ github.event.pull_request.head.sha || github.sha }})" >> $GITHUB_ENV
       - name: set $URL_STORYBOOK
+        if: steps.changes.outputs.deploy == 'true'
         run: echo "URL_STORYBOOK="https://${SHA7}.ods.dev"" >> $GITHUB_ENV
       - name: set $COMMIT_MSG
-        run: |
-          echo "COMMIT_MSG<<EOF" >> $GITHUB_ENV
-          git log --format=%B -n 1 ${{ github.event.after }} >> $GITHUB_ENV
-          echo 'EOF' >> $GITHUB_ENV
+        if: steps.changes.outputs.deploy == 'true'
+        run: echo "COMMIT_MSG=${{ github.event.head_commit.message || github.event.pull_request.title }}" >> $GITHUB_ENV
       - name: build storybook
-        run: |
-          yarn workspace @okta/odyssey-react build-storybook
+        if: steps.changes.outputs.deploy == 'true'
+        run: yarn workspace @okta/odyssey-react build-storybook --quiet
       - name: create deployment
+        if: steps.changes.outputs.deploy == 'true'
         uses: bobheadxi/deployments@v0.4.3
         id: deployment
         with:
           step: start
           token: ${{ secrets.GITHUB_TOKEN }}
           env: preview
       - name: deploy
-        run: |
-          aws s3 sync ./packages/odyssey-react/storybook-static/ s3://ods.dev/$SHA7 --delete
+        if: steps.changes.outputs.deploy == 'true'
+        run: aws s3 sync ./packages/odyssey-react/storybook-static/ s3://ods.dev/$SHA7 --delete
       - name: "inform slack of deployment (#odyssey-deployments)"
+        if: steps.changes.outputs.deploy == 'true'
         run: bash ./scripts/notify-slack.sh
         env:
           INCOMING_WEBHOOK_URL: ${{ secrets.SLACK_WEBHOOK_URL }}
           AUTHOR_NAME: ${{github.actor}}
           BRANCH_NAME: ${{github.head_ref}}
           PULL_REQUEST_ID: ${{github.event.pull_request.number}}
       - name: update deployment status
+        if: steps.changes.outputs.deploy == 'true'
         uses: bobheadxi/deployments@v0.4.3
-        if: always()
         with:
           step: finish
           token: ${{ secrets.GITHUB_TOKEN }}
           status: ${{ job.status }}
           deployment_id: ${{ steps.deployment.outputs.deployment_id }}
           env_url: ${{ env.URL_STORYBOOK }}
-  vrt:
-    needs: [prepare, test, deploy]
-    runs-on: ubuntu-latest
-    if: ${{ needs.prepare.outputs.vrt == 'true' }}
-    steps:
-      - name: checkout
-        uses: actions/checkout@v2
-      - name: setup node
-        uses: actions/setup-node@v2
-      - name: install dependencies
-        run: yarn install --frozen-lockfile --prefer-offline
-      - name: eyes-storybook (applitools)
-        run: yarn workspace @okta/odyssey-react eyes-storybook --exitcode=false
+
+      - name: visual regression test
+        if: steps.changes.outputs.vrt == 'true'
+        env:
+          APPLITOOLS_API_KEY: ${{ secrets.APPLITOOLS_API_KEY }}
+          APPLITOOLS_BATCH_ID: ${{ github.event.pull_request.head.sha }}
```

Comment:
only expose APPLITOOLS credentials to the step that needs than rather than all jobs etc